test(linter/no-console): add more tests

overlookmotel opened the pull request:
Add a few tests to `no-console` rule, to make sure it catches e.g. `console['log'](foo)`.

The test for ``console[`lo\\x67`](foo)`` ensures the correct operation of `ComputedMemberExpression::static_property_name` - that it uses `cooked` not `raw` for `TemplateLiteral`s.

camc314 commented:
### Merge activity

* **Jun 24, 12:55 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Jun 24, 12:56 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

overlookmotel commented:
Actually this doesn't test what I wanted it to. No matter how the string is interpreted (cooked or raw), it'll still be a fail. ``console[`lo\\x67`](foo)`` is a fail, same as `console[`batman`](foo)` is a fail - because any property of `console` is disallowed by default.

overlookmotel commented:
I think fixed now. This test fails if switch to using `MemberExpression::static_property_info`, which does have a bug (which I'll fix in next PR).

overlookmotel committed:
test(linter/no-console): add more tests (#11878)

Add a few tests to `no-console` rule, to make sure it catches e.g. `console['log'](foo)`.

The test for ``console[`lo\\x67`](foo)`` ensures the correct operation of `ComputedMemberExpression::static_property_name` - that it uses `cooked` not `raw` for `TemplateLiteral`s.
